drivers/mil_usb_to_can/mil_usb_to_can/device.py

```diff
@@ -35,6 +37,74 @@ def send_data(self, data, can_id=0):
         return self._driver.send_data(data, can_id=can_id)
 
 
+class ActuatorAddressOutOfRangeException(Exception):
+    '''
+    Exception thrown when the set_actuator service is called with an invalid address
+    '''
+    def __init__(self, address):
+        super(ActuatorAddressOutOfRangeException, self).__init__(
+            "actuator_address must be any integer from 0 to 11. The given address was {}".format(address))
+
+
+class ActuatorDeviceHandle(CANDeviceHandle):
+    '''
+    An implementation of a CANDeviceHandle which will handle
+    a device that controls pneumatic actuators.
+    '''
+
+    IDENTIFIER_BYTE = 'A'
+    NUM_ADDRESSES = 0x0B
+
+    def __init__(self, *args, **kwargs):
+        super(ActuatorDeviceHandle, self).__init__(*args, **kwargs)
+        self._set_srv = rospy.Service('~set_actuator', SetActuator, self.on_set_req)
+        self._get_srv = rospy.Service('~get_actuator', GetActuator, self.on_get_req)
+        self._actuator_on = [False for i in range(self.NUM_ADDRESSES)]
+        self._data_lock = [threading.Condition() for i in range(self.NUM_ADDRESSES)]
+
+    def on_set_req(self, req):
+        if (
+            not isinstance(req.actuator_address, int) or
+            req.actuator_address < 0 or
+            req.actuator_address > self.NUM_ADDRESSES
+        ):
+            raise ActuatorAddressOutOfRangeException(req.actuator_address)
+        message = self.IDENTIFIER_BYTE + chr(req.actuator_address) + chr(1)
+        if req.actuator_on:
+            message += chr(1)
+        else:
+            message += chr(0)
+        self.send_data(message)
+        return reduce((lambda a, b: a * pow(2, 8) + b), [ord(char) for char in message])
+
+    def on_get_req(self, req):
+        if not isinstance(req.actuator_address, int) or req.actuator_address < 0 or req.actuator_address > 0x0B:
```

Comment:
You don't really need to type check ROS messages, they are always serialized with the correct type